<launch>

  <param name="scan_line" type="int" value="16" />

  <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
  <param name="mapping_skip_frame" type="int" value="1" />

  <!-- remove too closed points -->
  <param name="minimum_range" type="double" value="0.25"/>


  <param name="mapping_line_resolution" type="double" value="0.2"/>
  <param name="mapping_plane_resolution" type="double" value="0.4"/>

  <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />

  <node pkg="tf" type="static_transform_publisher" name="camera_init_to_velodyne" args="0 0 0 0 0 0 camera_init velodyne 100" />

  <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen" />

  <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="screen" />

  <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="screen" />



  <node pkg="aloam_velodyne" name="adaptive_clustering" type="adaptive_clustering">
    <param name="print_fps" value="true"/>
  </node>
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/adaptive_clustering.rviz"/>
</launch>